geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -1193,10 +1195,15 @@ public void postCreateRegion() {
            * get the ParallelGatewaySender to create the colocated partitioned region for this
            * region.
            */
+          InternalGatewaySender senderImpl = (InternalGatewaySender) sender;
           if (sender.isRunning()) {
-            AbstractGatewaySender senderImpl = (AbstractGatewaySender) sender;
             ((ConcurrentParallelGatewaySenderQueue) senderImpl.getQueues()
                 .toArray(new RegionQueue[1])[0]).addShadowPartitionedRegionForUserPR(this);
+          } else if (senderImpl.getEventProcessor() == null) {
+            if (senderImpl.getStartupAction() == GatewaySenderStartupAction.STOP
+                || sender.isManualStart()) {
```

Comment:
The conditionals of this else-if block are kind of convoluted. I recommend extracting it to a single method with a name that describes it well and returns a single boolean so that you can rewrite it as:
```
} else if (isManualStartupRequired()) {
  senderImpl.recoverInStoppedState();
}
```
I'm not sure if `isManualStartupRequired()` is a good name... I'm just guessing here.